### PR TITLE
Publish monorepo-shipit

### PR DESCRIPTION
**Pull request by itsdouges**
This pull request updates monorepo-shipit to be able to be published to NPM and used by anyone. It:

- introduces new command line args
- removes hardcoded accounts
- updates internal workflow runs to use the new args

There is an opportunity in a follow up to replace the command line library with something like commanderjs so there is an easier way to look for available args, etc.

Related to #6096. Need to test it end-to-end before closing the issue.

### CODE REVIEW

**Comment by itsdouges** [comment]
@mrtnzlml hey mate this starts the process. I can't seem to get tests to pass, running on `master` they're also failing:

```
Seed:        748660241
Test Suites: 1 failed, 1 total
Tests:       5 failed, 5 total
Snapshots:   0 total
Time:        0.636 s, estimated 1 s
Ran all test suites matching /src\/__tests__\/RepoGit.findLastSourceCommit.test.js/i.

Active Filters: filename /src/__tests__/RepoGit.findLastSourceCommit.test.js/
```

Any ideas?

**Comment by mrtnzlml** [comment]
> @mrtnzlml hey mate this starts the process. I can't seem to get tests to pass, running on `master` they're also failing:
> 
> ```
> Seed:        748660241
> Test Suites: 1 failed, 1 total
> Tests:       5 failed, 5 total
> Snapshots:   0 total
> Time:        0.636 s, estimated 1 s
> Ran all test suites matching /src\/__tests__\/RepoGit.findLastSourceCommit.test.js/i.
> 
> Active Filters: filename /src/__tests__/RepoGit.findLastSourceCommit.test.js/
> ```
> 
> Any ideas?

@itsdouges This is unrelated to this PR. I've also noticed it locally (doesn't happen on CI though 🤔). I need to refactor/fix the test. Feel free to ignore it.

File: src/monorepo-shipit/package.json
Comment (mrtnzlml) [inline src/monorepo-shipit/package.json:1]: @itsdouges To comply with this monorepo requirements, could you please apply these changes to this file?

```patch
Index: src/monorepo-shipit/package.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/monorepo-shipit/package.json b/src/monorepo-shipit/package.json
--- a/src/monorepo-shipit/package.json	(revision Staged)
+++ b/src/monorepo-shipit/package.json	(date 1684432989334)
@@ -1,7 +1,16 @@
 {
   "name": "@adeira/monorepo-shipit",
+  "description": "Monorepo → many repos (Git) exporter.",
+  "homepage": "https://github.com/adeira/monorepo-shipit",
+  "bugs": "https://github.com/adeira/universe/issues",
+  "repository": {
+    "type": "git",
+    "url": "git@github.com:adeira/universe.git",
+    "directory": "src/monorepo-shipit"
+  },
   "license": "MIT",
   "version": "0.1.0",
+  "type": "commonjs",
   "bin": {
     "monorepo-importit": "bin/importit.js",
     "monorepo-shipit": "bin/shipit.js"
```

It has to be set for every package being published to NPM.

After that please run `yarn install` again, which should corretly update the `yarn.lock`.

File: src/monorepo-shipit/src/RepoGitFake.js
Comment (mrtnzlml) [inline src/monorepo-shipit/src/RepoGitFake.js:20]: Perhaps this username could be inlined as well now?

File: src/monorepo-shipit/bin/shipit.js
Comment (mrtnzlml) [inline src/monorepo-shipit/bin/shipit.js:33]: I think this is a great idea. Could you please update a README.md to recommend this approach? I think we should update README in general to explain the new `--committer-name` and `--committer-email` options (usage in general).

**Review by mrtnzlml — COMMENTED** [review]
Looks awesome! Much appreciated! 👏 

I left some minor comments. Basically the only important one is to update the `package.json` and we are good to go. 🚀

**Comment by mrtnzlml** [comment]
@itsdouges I saw that everything is passing, but the CI is currently not running for external contributors. That's on me: https://github.com/adeira/universe/issues/6110 🙈

**Comment by mrtnzlml** [comment]
@itsdouges I've fixed all the issues you reported (hopefully 🤞). Rebasing this PR should help.

**Comment by itsdouges** [comment]
Alright great thanks mate will take a look when I can.

File: src/monorepo-shipit/src/RepoGitFake.js
Comment (itsdouges) [reply src/monorepo-shipit/src/RepoGitFake.js]: Done

File: src/monorepo-shipit/package.json
Comment (itsdouges) [reply src/monorepo-shipit/package.json:1]: Done

File: src/monorepo-shipit/bin/shipit.js
Comment (itsdouges) [reply src/monorepo-shipit/bin/shipit.js:33]: Done. To swing back on committer-name and committer email - is this in essence what will commit to the destination repo, but not the author? 

So take your Flow example you linked

image

Us setting committer-name/email is in essence setting facebook-github-bot, yeah?

**Comment by itsdouges** [comment]
Ready for a review. Is there any way you can test this locally before we merge just in case? Would love to make sure nothing is broken.

Once it's merged and release I'll wire it up to my repository, will be at https://github.com/try-triplex/triplex

File: src/monorepo-shipit/bin/shipit.js
Comment (mrtnzlml) [reply src/monorepo-shipit/bin/shipit.js:33]: That is correct. The author is preserved, and the committer config is for `facebook-github-bot`. Here is another example where the committer is `adeira-github-bot`: https://github.com/adeira/sx/commit/10760b1ee4830b0a2e028c501157163ca68b83c4

**Review by mrtnzlml — APPROVED** [review]

**Comment by mrtnzlml** [comment]
> TBH the whole readme could have a once over to try and simplify first glance

Agree!

> There is an opportunity in a follow up to replace the command line library with something like commanderjs so there is an easier way to look for available args, etc.

Go for it if you want. It does the job, though, so consider whether you want to invest the energy there (GitHub integration for Shipit would be more valuable imho). 😎

**Comment by mrtnzlml** [comment]
I am about to merge this PR. Let's see if it all works and fix it in follow-up PRs if needed.

**Comment by mrtnzlml** [comment]
@itsdouges Thank you for your contribution. Well done! 👏

File: src/monorepo-shipit/bin/shipit.js
Comment (mrtnzlml) [inline src/monorepo-shipit/bin/shipit.js:51]: @itsdouges I missed this one. It's incorrectly flipped, right?

Comment (mrtnzlml) [reply src/monorepo-shipit/bin/shipit.js:51]: https://github.com/adeira/universe/pull/6131

Comment (itsdouges) [reply src/monorepo-shipit/bin/shipit.js:51]: Yes that's incorrect, I missed it too great catch

**Comment by itsdouges** [comment]
Sweet sounds great mate. I'll test it on my end too. See you in the original issue!

**Pull request merged**
